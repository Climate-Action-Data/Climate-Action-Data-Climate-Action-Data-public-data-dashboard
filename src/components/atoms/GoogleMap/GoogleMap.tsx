import { Box } from '@chakra-ui/react'
import { useEffect, useRef } from 'react'

interface GoogleMapProps {
  center: google.maps.LatLngLiteral
  zoom: number
}

export const GoogleMap = (props: GoogleMapProps) => {
  const { center, zoom } = props
  const ref = useRef<HTMLDivElement | null>(null)

  useEffect(() => {
    if (ref.current) {
      const map = new window.google.maps.Map(ref.current, {
        center,
        zoom,
      })

      const marker = new google.maps.Marker({
        position: center,
      })

      marker.setMap(map)
    }
  })

  return <Box borderTopLeftRadius="8px" borderTopRightRadius="8px" h="448px" w="448px" ref={ref} id="map" />
}
