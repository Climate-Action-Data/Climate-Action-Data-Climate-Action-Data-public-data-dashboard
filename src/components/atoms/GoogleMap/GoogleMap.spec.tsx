import { render } from '@testing-library/react'
import { GoogleMap } from './GoogleMap'

const DEFAULT_MAP_LOCATION = { lat: 45.76342, lng: 4.834277 }
const DEFAULT_MAP_ZOOM_IN = 10

const mockGeocoder = jest.fn((request, callback) => {
  callback([], `OK`)
})

const setupGoogleMock = () => {
  global.window.google = {
    maps: {
      Geocoder: jest.fn(() => ({
        geocode: mockGeocoder,
      })),
      Map: jest.fn(() => ({
        setCenter: jest.fn(),
        setZoom: jest.fn(),
      })),
      Marker: jest.fn(() => ({
        setMap: jest.fn(),
      })),
      GeocoderStatus: {
        ERROR: `ERROR`,
        INVALID_REQUEST: `INVALID_REQUEST`,
        OK: `OK`,
        OVER_QUERY_LIMIT: `OVER_QUERY_LIMIT`,
        REQUEST_DENIED: `REQUEST_DENIED`,
        UNKNOWN_ERROR: `UNKNOWN_ERROR`,
        ZERO_RESULTS: `ZERO_RESULTS`,
      },
    } as any,
  }
}

beforeAll(() => {
  setupGoogleMock()
})

it(`renders correctly`, () => {
  const { container } = render(<GoogleMap center={DEFAULT_MAP_LOCATION} zoom={DEFAULT_MAP_ZOOM_IN} />)
  expect(container).toMatchSnapshot()
})
