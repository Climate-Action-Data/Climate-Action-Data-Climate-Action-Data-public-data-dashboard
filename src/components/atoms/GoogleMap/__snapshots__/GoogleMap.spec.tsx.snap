// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-15uy3nq"
    id="map"
  />
</div>
`;
